docs/sources/flow/reference/components/remote.http.md

```diff
@@ -0,0 +1,150 @@
+following:
+
+* When the component first loads.
+* Every time the components arguments get re-evaluated.
+* At the frequency specified by the `poll_frequency` argument.
+
+The poll is successful if the URL returns a `200 OK` response code. All other
+response codes are treated as errors and mark the component as unhealthy. After
+a successful poll, the response body from the URL is exported.
+
+[secret]: {{< relref "../../config-language/expressions/types_and_values.md#secrets" >}}
+
+## Blocks
+
+The following blocks are supported inside the definition of `remote.http`:
+
+Hierarchy | Block | Description | Required
+--------- | ----- | ----------- | --------
+client | [client][] | HTTP client settings when connecting to the endpoint. | no
+client > basic_auth | [basic_auth][] | Configure basic_auth for authenticating to the endpoint. | no
+client > authorization | [authorization][] | Configure generic authorization to the endpoint. | no
+client > oauth2 | [oauth2][] | Configure OAuth2 for authenticating to the endpoint. | no
+client > oauth2 > tls_config | [tls_config][] | Configure TLS settings for connecting to the endpoint. | no
+client > tls_config | [tls_config][] | Configure TLS settings for connecting to the endpoint. | no
+
+The `>` symbol indicates deeper levels of nesting. For example, `client >
+basic_auth` refers to an `basic_auth` block defined inside a `client` block.
+
+[client]: #client-block
+[basic_auth]: #basic_auth-block
+[authorization]: #authorization-block
+[oauth2]: #oauth2-block
+[tls_config]: #tls_config-block
+
+### client block
+
+The `client` block configures settings used to connect to the HTTP
+server.
+
+{{< docs/shared lookup="flow/reference/components/http-client-config-block.md" source="agent" >}}
+
+### basic_auth block
+
+The `basic_auth` block configures basic authentication to use when polling the
+configured URL.
+
+{{< docs/shared lookup="flow/reference/components/basic-auth-block.md" source="agent" >}}
+
+### authorization block
+
+The `authorization` block configures custom authorization to use when polling
+the configured URL.
+
+{{< docs/shared lookup="flow/reference/components/authorization-block.md" source="agent" >}}
+
+### oauth2 block
+
+The `oauth2` block configures OAuth2 authorization to use when polling the
+configured URL.
+
+{{< docs/shared lookup="flow/reference/components/oauth2-block.md" source="agent" >}}
+
+### tls_config block
+
+The `tls_config` block configures TLS settings for connecting to HTTPS servers.
+
+{{< docs/shared lookup="flow/reference/components/tls-config-block.md" source="agent" >}}
+
+## Exported fields
+
+The following fields are exported and can be referenced by other components:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`content` | `string` or `secret` | The contents of the file. | | no
+
+If the `is_secret` argument was `true`, `content` is a secret type.
```

Comment:
Where is this `is-secret` argument of which you speak?  Oh, is this supposed to be just `secret`?

Reply:
It's the last argument in the arguments block :)